app/src/main/java/com/zulip/android/models/Stream.java

```diff
@@ -120,9 +120,8 @@ public static Stream getByName(ZulipApp app, String name) {
             if (stream == null) {
                 Log.w("Stream.getByName",
                         "We received a stream message for a stream we don't have data for. Fake it until you make it.");
-                stream = new Stream(name);
-                app.getDao(Stream.class).createIfNotExists(stream);
-            }
+                // catch the new stream object with auto-generated id
+                stream = app.getDao(Stream.class).createIfNotExists(new Stream(name));            }
```

Comment:
Well, storing a made-up stream ID for it might also cause a unique constraint exception if/when the stream that actually has that ID on the backend shows up, right?  More rare, but still a problem.  Hrm.

It seems like there are 2 possible correct solutions here:
* The same approach the web client uses, namely that we fetch from the server the complete list of streams we're subscribed to on startup, and maintain that list over time using the events system, so that this condition never happens.
* We modified the way messages are sent down to clients to optionally include the Stream ID in the display_recipient field (or something), so that the correct stream ID would be available here.

What do you think we should do?